### PR TITLE
Isolate api call wrapping into closure with connection

### PR DESCRIPTION
**Pull request by tshemsedinov**
It looks like dirty hack, but node.js do exactly same thing

### CODE REVIEW

File: lib/impress.jstp.js [hunk 1]
```diff
@@ -63,8 +63,7 @@ impress.jstp.mixin = (application) => {
 
     const startTime = process.hrtime();
 
-    application.sandbox.connection = connection;
-    method(...args.concat([onComplete]));
+    method(connection)(...args.concat([onComplete]));
```
Comment (tshemsedinov) [inline lib/impress.jstp.js]: @aqrln Here you can inject connection

File: lib/impress.application.js [hunk 2]
```diff
@@ -98,29 +98,31 @@ impress.application.mixin = (application) => {
     path.substr(application.dir.length)
   );
 
+  function apiWrap(source) {
+    return impress.USE_STRICT + '(connection => (' + source + '))';
```
Comment (tshemsedinov) [inline lib/impress.application.js]: @aqrln wrapping here

File: applications/example/config/passport.js [hunk 3]
```diff
@@ -1,40 +0,0 @@
-if (api.passport) {
```
Comment (aqrln) [inline applications/example/config/passport.js]: Unrelated to this PR, must be a separate commit.

File: applications/example/lib/application.test.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+// Example: application internal API (accessible from any place on application)
```
Comment (aqrln) [inline applications/example/lib/application.test.js]: Ditto.

File: lib/impress.application.js [hunk 5]
```diff
@@ -187,7 +189,7 @@ impress.application.mixin = (application) => {
   }
 
   application.createSandbox = (callback) => {
-    const sandbox = { module: {}, api: {} };
```
Comment (aqrln) [inline lib/impress.application.js]: I guess the deprecation of `__filename`, `__dirname` and `module` should be documented in, at least, commit message.

File: applications/example/config/passport.js (hunk 3 above)
Comment (tshemsedinov) [inline applications/example/config/passport.js]: It's unrelated but will not work after commit merged, ok, I'll do 2 commits in this PR

File: lib/impress.constants.js [hunk 6]
```diff
@@ -21,7 +21,7 @@ impress.FILE_PARSE_ERROR = 13;
 impress.DIR_EXISTS = 50;
 impress.DIR_NOT_EXISTS = 51;
 
-impress.USE_STRICT = '"use strict";\n';
+impress.USE_STRICT = '\'use strict\';\n';
```
Comment (aqrln) [inline lib/impress.constants.js]: Unrelated style change, but I'm fine with it being in this PR. I'd prefer it to be a second commit nevertheless, though.

File: applications/example/config/passport.js (hunk 3 above)
Comment (aqrln) [inline applications/example/config/passport.js]: Ah, I see. Well, then I'm fine with it in this commit.

Comment (aqrln) [inline applications/example/config/passport.js]: But the rationale of this change should be described in the full commit message.

File: lib/impress.jstp.js (hunk 1 above)
Comment (aqrln) [inline lib/impress.jstp.js]: Clever trick 👍

File: lib/impress.application.js (hunk 5 above)
Comment (tshemsedinov) [inline lib/impress.application.js]: Yes, it deprecates `module.exports` too because we have different modularity style, so I think no need to support node.js style or strange things like `__filename`, `__dirname` (may affect security too) and because applied scripts are supposed to know where they placed.

File: lib/impress.constants.js (hunk 6 above)
Comment (tshemsedinov) [inline lib/impress.constants.js]: Maybe even ``` `'use strict';\n` ```

Comment (aqrln) [inline lib/impress.constants.js]: SGTM if the linter won't complain.

**Comment by tshemsedinov** [comment]
Landed in 6516b523657311f01badcfe9c5bb36fcba789bee and bedac5952920ae835870f6e06381929c0e9c5754
